streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -293,14 +294,36 @@ public void commitOffsets() {
                 consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset));
                 stateMgr.putOffsetLimit(partition, offset);
             }
-            consumer.commitSync(consumedOffsetsAndMetadata);
+            try {
+                consumer.commitSync(consumedOffsetsAndMetadata);
+            } catch (CommitFailedException cfe) {
```

Comment:
Nit: add `final`
(please apply this wherever possible -- of course only code you change :) )